gateway/src/main/java/io/zeebe/gateway/grpc/ErrorMappingStreamObserver.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.gateway.grpc;
+
+import io.grpc.StatusException;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.ServerCallStreamObserver;
+import io.grpc.stub.StreamObserver;
+import io.zeebe.gateway.Loggers;
+
+/**
+ * A {@link StreamObserver} decorator which will map errors to {@link StatusException} before
+ * passing them on to the {@code delegate}.
+ *
+ * <p>Additionally, if the decorated {@link StreamObserver} is a {@link ServerCallStreamObserver},
```

Comment:
Actually you only accept ServerCallStreamObserver

Reply:
True, I wrote this before I changed the interface 😅